lib/changefeed.js

```diff
@@ -0,0 +1,154 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2020 Joyent, Inc.
+ * Copyright 2020 Bruce Smith.
+ */
+
+/*
+ * Exposes the internal change feed
+ */
+
+const EventEmitter = require('events');
+
+const assert = require('assert-plus');
+const changefeed = require('changefeed');
+const Watershed = require('watershed').Watershed;
+
+const shed = new Watershed();
+
+const translate = require('./machines.js').translate;
+
+class Feed extends EventEmitter {
+    constructor() {
+        super();
+    }
+
+    handler(req, res, next) {
+        this.latestMsgTimestamp = null;
+        // Start streaming in events from VMAPI
+        this.vmapiFeed = changefeed.createListener({
+            backoff: {
+                maxTimeout: Infinity,
+                minTimeout: 10,
+                retries: Infinity
+            },
+            log: req.log.child({
+                component: 'changefeed',
+                // Otherwise you'll see a ton of noise from here
+                level: 'fatal'
+            }),
+            url: req.config.vmapi.url,
+            instance: req.config.instanceUuid,
```

Comment:
What happens when a second changefeed request is performed whilst the first is still running (i.e. two `triton -i changefeed` commands)?

From my (limited) understanding - changefeed only allows one `instance` per `resource` to be listening, for which I have a suspicion that the first triton changefeed listener would no longer receive events after a second was registered...

Reply:
Indeed. See the comment regarding design flaws. Will workaround this creating a WebSocket server which publishes events as it receives them to the right channels, scoped by account uuid and optionally by VMs/vm

Reply:
Sweet - that'll fix it.